data-import/src/main/resources/folijet/data-import/features/data-import-integration.feature

```diff
@@ -2314,200 +2314,888 @@ Feature: Data Import integration tests
+                    }
+                ]
+            }
+        },
+        "addedRelations": [],
+        "deletedRelations": []
+    }
     """
     When method POST
-    Then status 204
-
-
-    ## verify job execution for data-import
-    * call pause 180000
-    * call read('classpath:folijet/data-import/features/get-completed-job-execution.feature@getJobWhenJobStatusCompleted') { jobExecutionId: '#(jobExecutionId)'}
-    * def jobExecution = response
-    And assert jobExecution.status == 'COMMITTED'
-    And assert jobExecution.uiStatus == 'RUNNING_COMPLETE'
-    And assert jobExecution.progress.current == 1
-    And assert jobExecution.progress.total == 1
-    And match jobExecution.runBy == '#present'
-    And match jobExecution.progress == '#present'
-
-
-    # verify that needed entities created
-    * call pause 10000
-    Given path 'metadata-provider/jobLogEntries', jobExecutionId
-    And headers headersUser
-    When method GET
-    Then status 200
-    And assert response.entries[0].sourceRecordActionStatus == 'CREATED'
-    And assert response.entries[0].instanceActionStatus == 'CREATED'
-    And match response.entries[0].error == '#notpresent'
-    * def sourceRecordId = response.entries[0].sourceRecordId
-
-
-    # retrieve instance hrid from record
-    Given path 'source-storage/records', sourceRecordId
-    And headers headersUser
-    When method GET
-    Then status 200
-    And match response.externalIdsHolder.instanceId == '#present'
-    * def instanceHrid = response.externalIdsHolder.instanceHrid
+    Then status 201
 
+    * def marcToHoldingsMappingProfileId = $.id
 
-    # verify that real instance was created with specific fields in inventory and retrieve instance id
-    Given path 'inventory/instances'
+    ## Create MARC-to-Item mapping profile
+    Given path 'data-import-profiles/mappingProfiles'
     And headers headersUser
-    And param query = 'hrid==' + instanceHrid
-    When method GET
-    Then status 200
-    And assert response.totalRecords == 1
-    And match response.instances[0].title == '#present'
-    And assert response.instances[0].identifiers[0].identifierTypeId == 'fcca2643-406a-482a-b760-7a7f8aec640e'
-    And assert response.instances[0].identifiers[0].value == '9780784412763'
-    And assert response.instances[0].identifiers[1].identifierTypeId == 'fcca2643-406a-482a-b760-7a7f8aec640e'
-    And assert response.instances[0].identifiers[1].value == '0784412766'
-    And assert response.instances[0].notes[0].instanceNoteTypeId == '86b6e817-e1bc-42fb-bab0-70e7547de6c1'
-    And assert response.instances[0].notes[0].note == 'Includes bibliographical references and index'
-    And assert response.instances[0].notes[0].staffOnly == false
-    And match response.instances[0].subjects contains  "Electronic books"
-    And match response.instances[0].subjects !contains "United States"
-
+    And request
+    """
+    {
+        "profile": {
```

Comment:
Is it possible to move profile jsons to separate files. In this case we will not mix them with source code and can reuse in another tests?

Reply:
@VRohach got it, agree. And what do you think regarding moving to separate json files to not mix with source code?

Reply:
It will be very difficult to reread the test scripts, opening the json files every time

Reply:
ok, got it